### PR TITLE
Update `winit` to `0.21`

### PR DESCRIPTION
**Pull request by Imberflur**
Builds on #147 by bumping to `0.21` and rebasing on `master`

### CODE REVIEW

File: winit/src/application.rs [hunk 1]
```diff
@@ -459,17 +474,7 @@ fn build_user_interface<'a, A: Application>(
...
-
-    (
-        physical_size.width.round() as u16,
-        physical_size.height.round() as u16,
-    )
-}
-
 // As defined in: http://www.unicode.org/faq/private_use.html
-// TODO: Remove once https://github.com/rust-windowing/winit/pull/1254 lands
 fn is_private_use_character(c: char) -> bool {
```
Comment (Imberflur) [inline winit/src/application.rs]: Should this be removed?

File: winit/src/application.rs [hunk 2]
```diff
@@ -337,18 +337,22 @@ pub trait Application: Sized {
...
                 WindowEvent::CloseRequested => {
                     *control_flow = ControlFlow::Exit;
                 }
                 WindowEvent::CursorMoved { position, .. } => {
+                    let position =
+                        position.to_logical::<f64>(size.scale_factor());
+
                     events.push(Event::Mouse(mouse::Event::CursorMoved {
                         x: position.x as f32,
                         y: position.y as f32,
```
Comment (Imberflur) [inline winit/src/application.rs]: Would is be desirable to mirror `winit` and use `f64` here?

File: winit/src/application.rs (hunk 1 above)
Comment (hecrj) [reply winit/src/application.rs]: I prefer to keep this for now. `winit` is not filtering exactly the same characters.

File: winit/src/application.rs (hunk 2 above)
Comment (hecrj) [reply winit/src/application.rs]: Probably, but we are currently using single precision almost everywhere.

I believe we need to think a bit more about it and eventually tackle it in a different PR.

**Comment by Imberflur** [comment]
Yay! It's great to see this merged :)

**Pull request merged**